test-data/unit/check-flags.test

```diff
@@ -509,3 +509,50 @@ x, y = 1, 2  # type: Unchecked, Unchecked
 [out]
 main:4: error: Type of variable becomes "Any" due to an unfollowed import
 main:6: error: A type on this line becomes "Any" due to an unfollowed import
+
+[case testDisallowAnyBasic]
+# flags: --disallow-any=expr
+from typing import List, Any, cast
+
+def foo(z: int):
+    return z
+
+def print(s: Any) -> None:
+    s = s + 1
+    pass
+
+class Foo:
+    g: Any = 2
+
+x: List[str] = ['hello']
+
+f: int = Foo.doo()
+k = Foo.doo()
+
+z = cast(int, Foo().g)
+m = cast(Any, Foo().g)
+y = Foo().g
+print(x[0])
+print(x[foo(0)])
+[builtins fixtures/list.pyi]
+[out]
+main:8: error: Expressions of type "Any" are disallowed (has type "Any")
```

Comment:
It'd be better to use the inline error syntax here (which you may not have seen).  Basically you can put `# E: ERROR MESSAGE` at the end of a line in a test, and it will be understood as indicating the appropriate error output on that line.